src/common/multi_file/multi_file_column_mapper.cpp

```diff
@@ -301,12 +452,16 @@ ColumnMapResult MapColumn(ClientContext &context, const MultiFileColumnDefinitio
 			// found a column mapping for this child - emplace it
 			column_mapping.emplace_back(global_child.name, std::move(child_map.column_map));
 		}
+		//! FIXME: the 'default_value' should only be used if the STRUCT's default value is not NULL
```

Comment:
After this merges I'll work on adding support for the `initial-default` to iceberg and build a test for the table of `point` `point.x` and `point.y` from here: https://iceberg.apache.org/spec/#default-values while likely making changes to core here to support these scenarios.